server/modules/survey/manager/surveyManager.js

```diff
@@ -136,7 +136,7 @@ export const fetchSurveyAndNodeDefsBySurveyId = async (
     fetchSurveyById(surveyId, draft, validate, client),
     NodeDefManager.fetchNodeDefsBySurveyId(surveyId, cycle, draft, advanced, includeDeleted, client),
   ])
-  const survey = Survey.assocNodeDefs(nodeDefs)(surveyDb)
+  const survey = R.pipe(Survey.assocNodeDefs(nodeDefs), Survey.buildAndAssocDependencyGraph)(surveyDb)
```

Comment:
why do you always buildAndAssocDependencyGraph when fetching survey and nodeDefs?

Reply:
right, it should be done only when "validate" is true